vmdb/app/models/mixins/authentication_mixin.rb

```diff
@@ -191,42 +191,44 @@ def authentication_check_types(*args)
     types.to_miq_a.each { |t| self.authentication_check(t, options)}
   end
 
+  # TODO: :valid, :incomplete, and friends shouldn't be littered in here and authentication
```

Comment:
Can you add a short description here about the return value?  Just want this to be uber clear throughout.  I'm not super crazy about the array return type, but I think it's fine as long as we have a quick blurb here about how it works.